tsl/src/bgw_policy/drop_chunks_api.c

```diff
@@ -91,11 +223,70 @@ drop_chunks_add_policy(PG_FUNCTION_ARGS)
 				 errmsg("could not add drop_chunks policy because \"%s\" is not a hypertable",
 						get_rel_name(ht_oid))));
 
+	/* validate that the open dimension uses a time type */
+	open_dim = hyperspace_get_open_dimension(hypertable->space, 0);
+	partitioning_type = ts_dimension_get_partition_type(open_dim);
+
+	switch (older_than_type)
+	{
+		case INTERVALOID:
+			if (IS_INTEGER_TYPE(partitioning_type))
+				ereport(ERROR,
+						(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+						 errmsg(
+							 "interval older_than argument can only be used with hypertables "
+							 "that have non-integer time dimensions"))); // <<-- time columns? open
+																		 // dimensions? TODO::
+			ts_dimension_open_typecheck(INTERVALOID, partitioning_type, "add_drop_chunks_policy");
+			older_than_interval = PG_GETARG_INTERVAL_P(1);
+			older_than_integer = -1;
+			break;
+		case INT2OID:
+		case INT4OID:
+		case INT8OID:
+			if (!IS_INTEGER_TYPE(partitioning_type))
+				ereport(ERROR,
+						(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+						 errmsg("integer older_than argument can only be used with hypertables "
+								"that have integer time dimensions"))); // <<-- time columns? open
+																		// dimensions? TODO::
+
+			/* TODO:: q:: awkward way to check if a NameStr col is null or not but did not find a
+			 * better API */
+			if ('\0' == *NameStr(open_dim->fd.integer_now_func) ||
```

Comment:
Is there a better way to know if a Name field is set or not?

Reply:
you should use strlen. It's fine to set this to an empty string if NULL;